bundles/org.openhab.binding.dbquery/src/main/java/org/openhab/binding/dbquery/internal/QueryHandler.java

```diff
@@ -0,0 +1,277 @@
+        } else {
+            logger.warn(
+                    "QueryResult discarded as queryResultChannelUpdater nor currentQueryExecution are not expected to be null");
+        }
+    }
+
+    private void updateCorrectChannel(boolean correct) {
+        updateState(DBQueryBindingConstants.CHANNEL_CORRECT, OnOffType.from(correct));
+    }
+
+    private void updateParametersChannel(QueryParameters queryParameters) {
+        updateState(DBQueryBindingConstants.CHANNEL_PARAMETERS, new StringType(jsonEncoder.encode(queryParameters)));
+    }
+
+    private void updateStateWithParentBridgeStatus() {
+        final @Nullable Bridge bridge = getBridge();
+        DatabaseBridgeHandler databaseBridgeHandler;
+
+        if (bridge != null) {
+            @Nullable
+            BridgeHandler bridgeHandler = bridge.getHandler();
+            if (bridgeHandler instanceof DatabaseBridgeHandler) {
+                databaseBridgeHandler = (DatabaseBridgeHandler) bridgeHandler;
+                database = databaseBridgeHandler.getDatabase();
+                if (bridge.getStatus() == ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+        }
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        super.updateStatus(status, statusDetail, description);
+        if (status == ThingStatus.ONLINE)
+            scheduleQueryExecutionIntervalIfNeeded();
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.trace("bridgeStatusChanged to {} for {}", bridgeStatusInfo.getStatus(), getQueryIdentifier());
+        cancelCurrentQueryExecution();
+        updateStateWithParentBridgeStatus();
+    }
+
+    public void setParameters(Map<String, @Nullable Object> parameters) {
+        final @Nullable QueryExecution queryExecution = currentQueryExecution;
+        if (queryExecution != null) {
+            QueryParameters queryParameters = new QueryParameters(parameters);
+            queryExecution.setQueryParameters(queryParameters);
+            queryExecution.execute();
+        } else {
+            logger.trace("setParameters ignored as there is any executing query for {}", getQueryIdentifier());
+        }
+    }
+
+    private final QueryExecution.QueryResultListener queryResultReceived = (QueryResult queryResult) -> {
+        synchronized (QueryHandler.this) {
+            logger.trace("queryResultReceived for {} : {}", getQueryIdentifier(), queryResult);
+            updateStateWithQueryResult(queryResult);
```

Comment:
Does this change the state of the thing or channel and get logged in the event log? if so please remove the logger.trace line.

Reply:
Leave as is.